### PR TITLE
Added check for downloading nasm

### LINKED ISSUE #1060: unexpected asgs build failures
I attempted a clean reinstall of ASGS on lonestar6 and my desktop machine, and both builds failed unexpectedly for different reasons. On lonestar6, it looks like the configure step for `ffmpeg` failed, with the error message as follows:
```
...
ffmpeg-4.4/presets/
ffmpeg-4.4/presets/libvpx-360p.ffpreset
ffmpeg-4.4/presets/libvpx-720p50_60.ffpreset
ffmpeg-4.4/presets/libvpx-1080p50_60.ffpreset
ffmpeg-4.4/presets/libvpx-720p.ffpreset
ffmpeg-4.4/presets/libvpx-1080p.ffpreset
ffmpeg-4.4/LICENSE.md
ffmpeg-4.4/fftools/
ffmpeg-4.4/fftools/ffmpeg_hw.c
ffmpeg-4.4/fftools/cmdutils.c
ffmpeg-4.4/fftools/Makefile
ffmpeg-4.4/fftools/ffmpeg_qsv.c
ffmpeg-4.4/fftools/ffmpeg_videotoolbox.c
ffmpeg-4.4/fftools/ffprobe.c
ffmpeg-4.4/fftools/ffmpeg_opt.c
ffmpeg-4.4/fftools/ffmpeg.h
ffmpeg-4.4/fftools/ffmpeg.c
ffmpeg-4.4/fftools/ffmpeg_filter.c
ffmpeg-4.4/fftools/cmdutils.h
ffmpeg-4.4/fftools/ffplay.c
ffmpeg-4.4/CONTRIBUTING.md
ffmpeg-4.4/README.md
nasm/yasm not found or too old. Use --disable-x86asm for a crippled build.

If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.freenode.net.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
Makefile:177: /tests/Makefile: No such file or directory
make: *** No rule to make target '/tests/Makefile'.  Stop.
Makefile:177: /tests/Makefile: No such file or directory
make: *** No rule to make target '/tests/Makefile'.  Stop.
post condition for "Step for installing ffmpeg" FAILED, stopping. Please fix and rerun.
```
On my `desktop` platform, it appears that the compilation of `wgrib2` failed with the following error message:
```
...
gcc -c -I/home/jason/Campaigns/Development/2022/issue-1059/opt/include -I/home/jason/Campaigns/Development/2022/issue-1059/grib2/include -Wall -Wmissing-prototypes -Wold-style-definition -Werror=format-security -ffast-math -O3 -DGFORTRAN -fopenmp -I/home/jason/Campaigns/Development/2022/issue-1059/grib2/jasper-1.900.1/src/libjasper/include    AAIGlong.c
gcc -c -I/home/jason/Campaigns/Development/2022/issue-1059/opt/include -I/home/jason/Campaigns/Development/2022/issue-1059/grib2/include -Wall -Wmissing-prototypes -Wold-style-definition -Werror=format-security -ffast-math -O3 -DGFORTRAN -fopenmp -I/home/jason/Campaigns/Development/2022/issue-1059/grib2/jasper-1.900.1/src/libjasper/include    Config.c
Config.c: In function ‘wgrib2api_info’:
Config.c:240:49: error: expected ‘;’ before ‘CC’
  240 |   return WGRIB2_VERSION "  " BUILD_COMMENTS " " CC;
      |                                                 ^~
make[2]: *** [makefile:60: Config.o] Error 1
make[2]: Leaving directory '/home/jason/Campaigns/Development/2022/issue-1059/grib2/wgrib2'
make[1]: *** [makefile:833: wgrib2/wgrib2] Error 2
make[1]: Leaving directory '/home/jason/Campaigns/Development/2022/issue-1059/grib2'
make: *** [makefile:76: wgrib2] Error 2
command for "Step for wgrib2" exited with an error code of 2, stopping. Please fix and rerun.
```
I'm not aware of anything that has changed for these builds, so I'm not sure what could be going on here. Any thoughts?

### PR DESCRIPTION
**Pull request by jasonfleming**
resolves #1060

### CODE REVIEW

File: cloud/general/init-ffmpeg.sh [hunk 1]
```diff
@@ -42,13 +42,17 @@ cd $_ASGS_TMP
...
 fi
-
-rm -rf $NASM_DIR 2> /dev/null
-tar zxvf $NASM_TGZ
-cd $NASM_DIR
-./configure --prefix=$OPT
-make
-make install
+# check that there are no issues with downloading nasm
+if [[ -e ${NASM_TGZ} && -s ${NASM_TGZ} &&  "$(file nasm-2.15.05.tar.gz)" =~ "gzip compressed data" ]]; then
```
Comment (wwlwpd) [inline cloud/general/init-ffmpeg.sh:46]:
```suggestion
if [[ -e ${NASM_TGZ} && -s ${NASM_TGZ} &&  "$(file ${NASM_TGZ} )" =~ "gzip compressed data" ]]; then
```

Comment (wwlwpd) [reply cloud/general/init-ffmpeg.sh:46]: Did you mean this? Also, checkout we can suggest smallish fixes in the code review itselft, neat!

(press this button to do so, image)

**Review by wwlwpd — CHANGES_REQUESTED** [review]
Check out the change suggestion I made, if good you should be able to directly accept / apply it from here.

**Comment by jasonfleming** [comment]
Thank you for that fix @wwlwpd, not sure how the literal file name got in there, I must have cut and pasted it. :-)

**Comment by jasonfleming** [comment]
I agree the PR fix inline suggest/commit mechanism is a productivity booster!

**Comment by jasonfleming** [comment]
I think this is ready to go.

**Review by wwlwpd — APPROVED** [review]

**Pull request merged**